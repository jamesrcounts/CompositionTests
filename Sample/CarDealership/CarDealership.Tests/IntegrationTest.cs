using CompositionTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CarDealership.Tests
{
    [TestClass]
    public class IntegrationTest
    {
        [TestMethod]
        public void VerifyComposition()
        {
            MefComposition.VerifyAssemblyCatalog(Program.Catalog);
        }
    }
}