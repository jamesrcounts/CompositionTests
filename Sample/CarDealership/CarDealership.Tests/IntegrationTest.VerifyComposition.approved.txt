[Part] CarDealership.Dealership from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
  [Export] CarDealership.Dealership (ContractName="CarDealership.Dealership")
  [Import] CarDealership.Dealership.Cars (ContractName="CarDealership.Car")
    [SatisfiedBy] CarDealership.Ford (ContractName="CarDealership.Car") from: CarDealership.Ford from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
    [SatisfiedBy] CarDealership.Car (ContractName="CarDealership.Car") from: CarDealership.Ford from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
    [SatisfiedBy] CarDealership.Car (ContractName="CarDealership.Car") from: CarDealership.Honda from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
    [Unsuitable] CarDealership.Car (ContractName="CarDealership.Car") from: CarDealership.Mazda from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
      [Because] PartDefinitionIsRejected, The part providing the export is rejected because of other issues.

[Part] CarDealership.Ford from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
  [Export] CarDealership.Ford (ContractName="CarDealership.Car")
  [Export] CarDealership.Car (ContractName="CarDealership.Car")
  [Import] CarDealership.Ford.Motor (ContractName="CarDealership.IMotor")
    [SatisfiedBy] CarDealership.V8Motor (ContractName="CarDealership.IMotor") from: CarDealership.V8Motor from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")

[Part] CarDealership.Honda from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
  [Export] CarDealership.Car (ContractName="CarDealership.Car")

[Part] CarDealership.Mazda from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
  [Primary Rejection]
  [Export] CarDealership.Car (ContractName="CarDealership.Car")
  [Import] CarDealership.Mazda.Engine (ContractName="CarDealership.Motor")
    [Exception] System.ComponentModel.Composition.ImportCardinalityMismatchException: No exports were found that match the constraint: 
    ContractName    CarDealership.Motor
    RequiredTypeIdentity    CarDealership.Motor
   at System.ComponentModel.Composition.Hosting.ExportProvider.GetExports(ImportDefinition definition, AtomicComposition atomicComposition)
   at System.ComponentModel.Composition.Hosting.ExportProvider.GetExports(ImportDefinition definition)
   at Microsoft.ComponentModel.Composition.Diagnostics.CompositionInfo.AnalyzeImportDefinition(ExportProvider host, IEnumerable`1 availableParts, ImportDefinition id)

[Part] CarDealership.V8Motor from: AssemblyCatalog (Assembly="CarDealership, Culture=neutral")
  [Export] CarDealership.V8Motor (ContractName="CarDealership.IMotor")

